libraries/ms-drivers/src/adt7476a_fan_controller.c

```diff
@@ -21,7 +21,7 @@ StatusCode adt7476a_set_speed(I2CPort port, uint8_t speed_percent, AdtPwmPort pw
   }
 
   // determine which PWM output to change
-  uint8_t real_speed = (speed_percent / 0.39);
+  uint8_t real_speed = (speed_percent == 0) ? 0 : (speed_percent / 0.39 - 1);
```

Comment:
With previous, a real speed of 100 would give a value of 0x100 instead of the expected 0xff

Reply:
Nice catch! I wonder why the decision was made to use 0-100 instead of 0x00-0xff in the first place, necessitating this conversion.